### ISSUE TITLE
Intermittent failure: gather facts timeout when provisioning beaker

### ISSUE THREAD

**Issue by jaypoulz** [CentOS-PaaS-SIG/linchpin#690]
This is something I've been running into lately. Seems to be present in the 1.6.x releases, but I suspect it's been an issue for longer that that.

I can't reproduce it consistently, but when I do hit it, it produces the following error:

```
TASK [Gathering Facts] *********************************************************

task path: /home/jenkins/envs/provisioner/lib/python2.7/site-packages/linchpin/provision/beaker.yml:4

The full traceback is:

Traceback (most recent call last):

  File "/tmp/ansible_J25NCa/ansible_modlib.zip/ansible/module_utils/basic.py", line 2859, in run_command

    cmd = subprocess.Popen(args, **kwargs)

  File "/usr/lib64/python2.7/subprocess.py", line 711, in __init__

    errread, errwrite)

  File "/usr/lib64/python2.7/subprocess.py", line 1308, in _execute_child

    data = _eintr_retry_call(os.read, errpipe_read, 1048576)

  File "/usr/lib64/python2.7/subprocess.py", line 478, in _eintr_retry_call

    return func(*args)

  File "/tmp/ansible_J25NCa/ansible_modlib.zip/ansible/module_utils/facts/timeout.py", line 37, in _handle_timeout

    raise TimeoutError(msg)

TimeoutError: Timer expired after 10 seconds

fatal: [localhost]: FAILED! => {

    "changed": false, 

    "cmd": "/usr/sbin/udevadm info --query property --name /dev/vda1", 

    "invocation": {

        "module_args": {

            "fact_path": "/etc/ansible/facts.d", 

            "filter": "*", 

            "gather_subset": [

                "all"

            ], 

            "gather_timeout": 10

        }

    }, 

    "msg": "Timer expired after 10 seconds", 

    "rc": 257

}
```

**Comment by herlo** [comment]
@jaypoulz are you hitting this within a container? I suspect not.

My best guess for this is the gather facts is taking a long time to get data. There is at least one easy way to mitigate the issue.  In ansible.cfg, modify the following lines to match as below:

```
Line 48:  gather_subset = facter
Line 55:  gather_timeout = 30
```

Personally, I'd only do the first of the two edits. However, if the error creeps up again, extending that timeout may help. It will, however, possibly wait up to that number of seconds before timing out, which means the provisioning process is slower.

Hope this helps.

@herlo

**Comment by jaypoulz** [comment]
I am hitting this within a container. I think it's around a 30-40% hit rate at the moment. I will see what I can do about the ansible.cfg change.

**Comment by jaypoulz** [comment]
Looks like it may be related to:
https://github.com/ansible/ansible/issues/43884

**Comment by jaypoulz** [comment]
I'm going to close this issue since it's one where this underlying issue is with the Ansible 2.6.x series. I've verified that downgrading to 2.5.9 stabilized linchpin for me without further changes to the Ansible config.

**Comment by johnbieren** [comment]
ANSIBLE_GATHER_FACTS=25 fixed this for me with ansible 2.7
